pymc_experimental/tests/test_distributions.py

```diff
@@ -0,0 +1,50 @@
+#   Copyright 2020 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+# general imports
+import scipy.stats.distributions as ssd
+
+# test support imports from pymc
+from pymc.tests.test_distributions import (
+    R,
+    Rplus,
+    Domain,
+    TestMatchesScipy,
+)
+
+# the distributions to be tested
+from pymc_experimental.distributions import (
+    GenExtreme,
+)
+
+
+class TestMatchesScipyX(TestMatchesScipy):
```

Comment:
We removed the `check_logp`/`check_logcdf` into https://github.com/pymc-devs/pymc/blob/a38bead63f71e3dac1c589c73acb98174f3b8e59/pymc/tests/distributions/util.py#L230, so you no longer need to subclass the old `TestMatchesScipy`. Just use a `TestGenExtreme` class from scratch.

Reply:
Also, the tests in the main repo have been refactored, and now fall in the folder where the distribution is implemented, so in this case would be `/tests/distributions/test_continuous`

Reply:
Yes, I've moved all those functions now into one place.

However, the tests are failing. Parameters values outside of the set `Domain` are being generated for `xi`, and so it is failing at [this point](https://github.com/pymc-devs/pymc/blob/372d7c247ac55bd90ed999031a4e207fa24757be/pymc/tests/distributions/util.py#L359). I don't know why values of `xi=-2.0` are being proposed, for example. It is as if the new `check_logp` is not evaluating or respecting the set Domain? Example output:
```
Failed: test_params={'mu': array(-2.1), 'sigma': array(0.01), 'xi': TensorConstant{-2.0}}, valid_
value=-2.1
```
and yet when I implement my own test, it passes (code without obvious imports):
```python
μ,σ,ξ,x = -2.1,0.01,-2.0,-2.1
mu = at.as_tensor_variable(floatX(μ))
sigma = at.as_tensor_variable(floatX(σ))
xi = at.as_tensor_variable(floatX(ξ))
logp_pm = pmx.distributions.GenExtreme.logp(x,mu,sigma,xi).eval()
logp_sp = sp.genextreme.logpdf(x, c=-ξ, loc=μ, scale=σ)
if not np.allclose(logp_pm,logp_sp):
    print(f"Fail on {μ=}, {σ=}, {ξ=}")
```
So I need some input at this point for the tests - thanks!